src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -4457,6 +4457,44 @@ module TcDeclarations =
                 if not (List.isEmpty attributes) && (declKind = ExtrinsicExtensionBinding || declKind = IntrinsicExtensionBinding) then
                     let attributeRange = (List.head attributes).Range
                     error(Error(FSComp.SR.tcAugmentationsCannotHaveAttributes(), attributeRange))
+                    
+                let staticAbstractMembers =
+                    tcref.ImmediateInterfaceTypesOfFSharpTycon
+                    |> List.choose(
+                        fun ttype ->
+                            match stripTyEqnsAndMeasureEqns g ttype with
+                            | TType_app(tyconRef = tcref) ->
+                                let staticAbstractMembers =
+                                    tcref.MembersOfFSharpTyconSorted
+                                    |> List.filter(fun x -> not x.IsInstanceMember)
+                                Some staticAbstractMembers
+                            | _ -> None
+                    )
+                    |> List.concat
+                        
+                let implMembers =
+                    members
+                    |> List.choose(
+                        fun synMemberDef ->
```

Comment:
Wondering if those aren't simpler expressed as a nested for loop in a list expression; could also collapse the match and fun into function if it remains a pipe.